src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -3742,14 +3742,14 @@ PartitionColumnOpExpressionList(Query *query)
 			continue;
 		}
 
-		leftArgument = (Node *) linitial(argumentList);
-		rightArgument = (Node *) lsecond(argumentList);
+		leftArgument = strip_implicit_coercions((Node *) linitial(argumentList));
```

Comment:
Here, we do two things at one line;
i. We get the data in the first cell of a List
ii. We apply coercion strip

For coding convention and better readability, we prefer to do this kind of work on individual code lines. An example from the code base;

```
Node *leftOpExpression = linitial(arrayOperatorExpression->args);
Node *strippedLeftOpExpression = strip_implicit_coercions(leftOpExpression);
```